src/main/java/org/igniterealtime/openfire/plugins/pushnotification/PushInterceptor.java

```diff
@@ -97,17 +103,22 @@ public void interceptPacket( final Packet packet, final Session session, final b
             return;
         }
 
-        Log.trace( "If user '{}' has push services configured, pushes need to be sent for a message that just arrived.", user );
-        tryPushNotification( user );
+        Log.debug( "If user '{}' has push services configured, pushes need to be sent for a message that just arrived.", user );
+        tryPushNotification( user, body, packet.getFrom(), ((Message) packet).getType() );
     }
 
-    private void tryPushNotification( User user )
+    private void tryPushNotification( User user, String body, JID jid, Message.Type msgtype )
     {
+        if (XMPPServer.getInstance().getPresenceManager().isAvailable( user ))
+        {
+            return; // dont notify if user is online and available. let client handle that
+        }
+
```

Comment:
I think this is the cause of the problems that are being reported https://github.com/anurodhp/Monal/issues/354 (even though this PR has not been merged, it did generate a nightly that people, in lack of a proper release of this plugin, are using).

Although it makes sense to limit the amount of web pushes (that have quotes), this prevents push notifications for clients that have been silenced by the OS, without Openfire having picked up on that fact yet (SM might be in play, or it's still waiting for a timeout, etc).

We should find a way to limit push notifications only after we've determined that the to-be-generated notification is of the web-type. Or, potentially, we create a different implementation alltogether for web-based push notifications. Let's find a way that does force us to support either native or web notifications.